scripts/loadable_app/hello.c

```diff
@@ -0,0 +1,12 @@
+int a = 9;
+int b = 9;
+
+int x()
+{
+return 7;
+}
+int main(int argc, char **argv)
+{
+	int p = x();
+	return a + p + x();
```

Comment:
Sorry, github decided not to send notification for your reply to this comment

I made following steps
1. git fetch https://github.com/embox/embox pull/1090/head:pull-1090 // take pull request
1. git checkout pull-1090
1. make confload-x86/qemu
1. Change CROSS_COMPILE in conf/build.conf to point to https://github.com/embox/crosstool/releases/download/2.28-6.3.0-7.12/i386-elf-toolchain.tar.bz2 // I have gcc7 on host system, so build fails with warnings fixed in #1088 
1. add `load_app` to conf/mods.config
1. cd scripts/loadable_app/; ./loadable_app.sh; cd -; cp scripts/loadable_app/hello conf/rootfs/
1. make
1. qemu-system-i386 -kernel ./build/base/bin/embox -enable-kvm -m 256 -nographic
1. root@embox:/#load_app hello
```
load_app: Command returned with code 14: Unknown error code. PLEASE REPORT ISSUE 519!
tish error: #14
```
My host compiler is gcc (GCC) 7.1.1 20170528, but I've tried at first on different system with some gcc6.x.x, but they have no difference.
The host compiler could play a big difference here, I suppose. Your could give different result.

I've applied next for simplicity
```
--- a/src/cmds/execute.c
+++ b/src/cmds/execute.c
@@ -60,6 +60,8 @@ int main(int argc, char **argv) {
 
        int ret = functionPtr();
 
+       printf("ret=%d\n", ret);
+
        close(elf_file);
        free(header);
        free(ph_table);
```

If I change hello.c to 
```
int a = 17;
int main(int argc, char **argv) {
        return a;
}
```
dissassmbling of this (objdump -d hello) looks like
```
080480b8 <main>:
 80480b8:       55                      push   %ebp
 80480b9:       89 e5                   mov    %esp,%ebp
 80480bb:       e8 0f 00 00 00          call   80480cf <__x86.get_pc_thunk.ax>
 80480c0:       05 60 00 00 00          add    $0x60,%eax
 80480c5:       c7 c0 2c 81 04 08       mov    $0x804812c,%eax
 80480cb:       8b 00                   mov    (%eax),%eax
 80480cd:       5d                      pop    %ebp
 80480ce:       c3                      ret    

080480cf <__x86.get_pc_thunk.ax>:
 80480cf:       8b 04 24                mov    (%esp),%eax
 80480d2:       c3                      ret    
```

and the result is 
```
#load_app hello                                                     
ret=0
```
Take a note, that instruiction on 80480c5 have `$0x804812c` in it's encoding (`2c 81 04 08`), that' should mean it's absolute. I think, it really tries to load from 0x804812c

I tried to remove `-fpic` from loadable_app.sh, dissasembling looks like
```
080480b8 <main>:
 80480b8:       55                      push   %ebp
 80480b9:       89 e5                   mov    %esp,%ebp
 80480bb:       a1 fc 80 04 08          mov    0x80480fc,%eax
 80480c0:       5d                      pop    %ebp
 80480c1:       c3                      ret    
```

Result is unchanged, (`ret=0`), loading from absolute address still remain.

Could you provide your `hello` binary and it's dissasmbly, so I could try/look at it?